art/estimators/speech_recognition/pytorch_deep_speech.py

```diff
@@ -0,0 +1,636 @@
+                  `x = np.ndarray([[0.1, 0.2, 0.1, 0.4], [0.3, 0.1]])`.
+        :param batch_size: Batch size.
+        :param transcription_output: Indicate whether the function will produce probability or transcription as
+                                     prediction output.
+        :type transcription_output: `bool`
+        :param decoder_type: Decoder type. Either `greedy` or `beam`. This parameter is only used when users want
+                             transcription outputs. Default to `greedy`.
+        :type decoder_type: `str`
+        :param lm_path: Path to an (optional) kenlm language model for use with beam search. This parameter is only
+                        used when users want transcription outputs. Default to `''`.
+        :type lm_path: `str`
+        :param top_paths: Number of beams to return. This parameter is only used when users want transcription outputs.
+                          Default to 1.
+        :type top_paths: `int`
+        :param alpha: Language model weight. This parameter is only used when users want transcription outputs.
+                      Default to 0.0.
+        :type alpha: `float`
+        :param beta: Language model word bonus (all words). This parameter is only used when users want transcription
+                     outputs. Default to 0.0.
+        :type beta: `float`
+        :param cutoff_top_n: Cutoff_top_n characters with highest probs in vocabulary will be used in beam search. This
+                             parameter is only used when users want transcription outputs. Default to 40.
+        :type cutoff_top_n: `float`
+        :param cutoff_prob: Cutoff probability in pruning. This parameter is only used when users want transcription
+                            outputs. Default to 1.0.
+        :type cutoff_prob: `float`
+        :param beam_width: Beam width to use. This parameter is only used when users want transcription outputs.
+                           Default to 10.
+        :type beam_width: `int`
+        :param lm_workers: Number of language model processes to use. This parameter is only used when users want
+                           transcription outputs. Default to 4.
+        :type lm_workers: `int`
+        :return: Probability (if transcription_output is None or False) or transcription (if transcription_output is
+                 True) predictions:
+                    - Probability return is a tuple of (probs, sizes), where:
+                        - probs is the probability of characters of shape (nb_samples, seq_length, nb_classes).
+                        - sizes is the real sequence length of shape (nb_samples,).
+                    - Transcription return is a numpy array of characters. A possible example of a transcription return
+                      is `np.array(['SIXTY ONE', 'HELLO'])`.
+        """
+        import torch  # lgtm [py/repeated-import]
+
+        x_ = x.copy()
```

Comment:
Is this copy necessary?

Reply:
Yes this is necessary as the value of x pointers will be changed later in the transformer.